src/rgw/setup/rgw.py

```diff
@@ -327,15 +338,43 @@ def _check_consul_connection(conf: MappedConf):
             raise SetupError(errno.EINVAL, f"Consul server {host:port} not reachable")
 
     @staticmethod
-    def _fetch_endpoint_url(conf: MappedConf, confstore_endpoint_key: str, endpoint_type: str):
+    def _get_gconf_key_list(conf: MappedConf, gconf_num_key:str, actual_gconf_key:str):
+        """Get value list of specified gconf key."""
+        # e.g. for single key (key - cortx>common>external>consul>endpoints), it may have multiple values as,
+        # values: [tcp://cortx-consul-server:8301, http://cortx-consul-server:8301].
+        # To get this list of values,
+        #     a) first get number of endpoints for this key using another key (cortx>common>external>consul>num_endpoints),
```

Comment:
Am not sure why this num_endpoints is even introduced, given we can just iterate over configured number of endpoints? @tanujashinde0405 @nitin-seagate

Reply:
@cdeshmukh , num_ was added in gconf for all lists as requested by @ujjwalpl . Also As fas as i know, you cannot ietrate over multiple values in consul if you don't know how many values are there. Hence num_ keys helpes to iterate over lists .

Reply:
yes correct, earlier we use to pass key to yaml based Gconf, which was returning all available values as list in one command but here in consul based Gconf it does not return list hence we need to iterate over each index and get the value using key & index. 
To get this max no of index, we need to fetch this **num_** (num_endpoints, num_services) key in Gconf.

Reply:
When i look at consul cli interface, listing keys should help here, that way we dont need to depend on the number of entries, so we can improve i believe @tanujashinde0405 

https://www.consul.io/commands/kv/get#listing-keys
![image](https://user-images.githubusercontent.com/32053979/179160858-ec1dcc35-83b5-4292-a2aa-e6da2c568d7c.png)


Anyways, there is no pointing holding on to this PR for GConf improvements